examples/calibration/plot_calibration.py

```diff
@@ -0,0 +1,116 @@
+"""
+======================================
+Probability calibration of classifiers
+======================================
+
+When performing classification you often want to predict not only
+the class label, but also the associated probability. This probability
+gives you some kind of confidence on the prediction. However, not all
+classifiers provide well-calibrated probabilities, some being over-confident
+while others being under-confident. Thus, a separate calibration of predicted
+probabilities is often desirable as a postprocessing. This example illustrates
+two different methods for this calibration and evaluates the quality of the
+returned probabilities using Brier's score
+(see http://en.wikipedia.org/wiki/Brier_score).
+
+Compared are the estimated probability using a Gaussian naive Bayes classifier
+without calibration, with a sigmoid calibration, and with a non-parametric
+isotonic calibration. One can observe that only the non-parametric model is able
+to provide a probability calibration that returns probabilities close to the
+expected 0.5 for most of the samples belonging to the middle cluster with
+heterogeneous labels. This results in a significantly improved Brier score.
+"""
+print(__doc__)
+
+# Author: Mathieu Blondel <mathieu@mblondel.org>
+#         Alexandre Gramfort <alexandre.gramfort@telecom-paristech.fr>
+#         Balazs Kegl <balazs.kegl@gmail.com>
+#         Jan Hendrik Metzen <jhm@informatik.uni-bremen.de>
+# License: BSD Style.
+
+import numpy as np
+import matplotlib.pyplot as plt
+from matplotlib import cm
+
+from sklearn.datasets import make_blobs
+from sklearn.naive_bayes import GaussianNB
+from sklearn.metrics import brier_score_loss
+from sklearn.calibration import CalibratedClassifierCV
+from sklearn.cross_validation import train_test_split
+
+
+n_samples = 50000
+n_bins = 3  # use 3 bins for calibration_curve as we have 3 clusters here
+
+# Generate 3 blobs with 2 classes where the second blob contains
+# half positive samples and half negative samples. Probability in this
+# blob is therefore 0.5.
+centers = [(-5, -5), (0, 0), (5, 5)]
+X, y = make_blobs(n_samples=n_samples, n_features=2, cluster_std=1.0,
+                  centers=centers, shuffle=False, random_state=42)
+
+y[:n_samples // 2] = 0
+y[n_samples // 2:] = 1
+sample_weight = np.random.RandomState(42).rand(y.shape[0])
+
+# split train, test for calibration
+X_train, X_test, y_train, y_test, sw_train, sw_test = \
+    train_test_split(X, y, sample_weight, test_size=0.9, random_state=42)
+
+# Gaussian Naive-Bayes with no calibration
+clf = GaussianNB()
+clf.fit(X_train, y_train)  # GaussianNB itself does not support sample-weights
+prob_pos_clf = clf.predict_proba(X_test)[:, 1]
+
+# Gaussian Naive-Bayes with isotonic calibration
+clf_isotonic = CalibratedClassifierCV(clf, cv=2, method='isotonic')
+clf_isotonic.fit(X_train, y_train, sw_train)
```

Comment:
i think we agreed that throwing a warning makes sense here: we can use the sample-weights during the calibration but not during the fitting of the base-classifier. We could catch the warning here but I agree with @agramfort that adding sample_weights to GaussianNB would be nicer.